cli/azd/pkg/infra/provisioning/bicep/bicep_provider_test.go

```diff
@@ -974,7 +975,39 @@ func TestUserDefinedTypes(t *testing.T) {
 			"sku":  {Type: "string"},
 		},
 		objectParam.Properties)
+	require.NotNil(t, objectParam.AdditionalProperties)
+	require.Equal(
+		t,
+		azure.ArmTemplateParameterAdditionalProperties{
+			Type:      "string",
+			MinLength: to.Ptr(10),
+			Metadata: map[string]json.RawMessage{
+				"fromDefinitionFoo": []byte(`"foo"`),
+				"fromDefinitionBar": []byte(`"bar"`),
+			},
+		},
+		objectParam.AdditionalProperties)
+	require.NotNil(t, objectParam.Metadata)
+	require.Equal(
+		t,
+		map[string]json.RawMessage{
+			// Note: Validating the metadata combining and override here.
+			// The parameter definition contains metadata that is automatically added to the parameter.
+			// Then the parameter also has metadata and overrides one of the values from the definition.
+			"fromDefinitionFoo": []byte(`"foo"`),
+			"fromDefinitionBar": []byte(`"override"`),
+			"fromParameter":     []byte(`"parameter"`),
+		},
+		objectParam.Metadata)
 
+	// output resolves just the type. Value and Metadata should persist
+	customOutput, exists := template.Outputs["customOutput"]
+	require.True(t, exists)
+	require.Equal(t, "string", customOutput.Type)
+	require.Equal(t, "[parameters('stringLimitedParam')]", customOutput.Value)
+	require.Equal(t, map[string]interface{}{
+		"foo": "bar",
+	}, customOutput.Metadata)
```

Comment:
What is the final output that is saved into the `.env` file?

Reply:
We only use the output name and value. We don't use `metadata` or any other property from an output